File: app/src/androidTest/java/za/co/riggaroo/datecountdown/util/CountdownTestRunner.java
package za.co.riggaroo.datecountdown.util;

import android.app.Application;
import android.content.Context;
import android.support.test.runner.AndroidJUnitRunner;

import za.co.riggaroo.datecountdown.TestApplication;

public class CountdownTestRunner extends AndroidJUnitRunner {
    @Override
    public Application newApplication(ClassLoader cl, String className, Context context)
            throws InstantiationException, IllegalAccessException, ClassNotFoundException {
        return super.newApplication(cl, TestApplication.class.getName(), context);
    }
}

File: app/src/androidTest/java/za/co/riggaroo/datecountdown/util/CountdownTestRunner.kt
package za.co.riggaroo.datecountdown.util

import android.app.Application
import android.content.Context
import android.support.test.runner.AndroidJUnitRunner
import com.squareup.rx2.idler.Rx2Idler
import io.reactivex.plugins.RxJavaPlugins
import za.co.riggaroo.datecountdown.CountdownApplication


class CountdownTestRunner : AndroidJUnitRunner() {

    override fun onStart() {
        RxJavaPlugins.setInitComputationSchedulerHandler(
                Rx2Idler.create("RxJava 2.x Computation Scheduler"))

        super.onStart()
    }

    @Throws(InstantiationException::class, IllegalAccessException::class, ClassNotFoundException::class)
    override fun newApplication(cl: ClassLoader, className: String, context: Context): Application {
        return super.newApplication(cl, CountdownApplication::class.java.name, context)
    }
}
